source/CScriptEngine.cpp

```diff
@@ -684,79 +684,64 @@ namespace CLEO
+        if(!fsPath.empty())
         {
-            auto fsPath = FS::path(path);
+            const auto root = fsPath.begin()->string(); // first path element
+            const auto r = root.c_str();
 
-            // check for virtual path root
-            enum class VPref{ None, Game, User, Script, Cleo, Modules } virtualPrefix = VPref::None;
-            auto root = fsPath.begin();
-            if(root != fsPath.end())
-            {
-                if(*root == DIR_GAME) virtualPrefix = VPref::Game;
-                else if (*root == DIR_USER) virtualPrefix = VPref::User;
-                else if (*root == DIR_SCRIPT) virtualPrefix = VPref::Script;
-                else if (*root == DIR_CLEO) virtualPrefix = VPref::Cleo;
-                else if (*root == DIR_MODULES) virtualPrefix = VPref::Modules;
-            }
+            if(_strcmpi(r, DIR_GAME) == 0) virtualPrefix = VPref::Game;
+            else if (_strcmpi(r, DIR_USER) == 0) virtualPrefix = VPref::User;
+            else if (_strcmpi(r, DIR_SCRIPT) == 0) virtualPrefix = VPref::Script;
+            else if (_strcmpi(r, DIR_CLEO) == 0) virtualPrefix = VPref::Cleo;
+            else if (_strcmpi(r, DIR_MODULES) == 0) virtualPrefix = VPref::Modules;
+        }
 
-            // not virtual
-            if(virtualPrefix == VPref::None)
+        // not virtual
+        if(virtualPrefix == VPref::None)
+        {
+            if(fsPath.is_relative())
             {
-                if(fsPath.is_relative())
-                {
-                    if(customWorkDir != nullptr)
-                        fsPath = ResolvePath(customWorkDir) / fsPath;
-                    else
-                        fsPath = GetWorkDir() / fsPath;
-
-                    auto resolved = FS::weakly_canonical(fsPath).string();
-
-                    // ModLoader support: do not expand game dir relative paths
-                    if (resolved.find(Filepath_Root) == 0)
-                        return FS::relative(resolved, Filepath_Root).string();
-                    else
-                        return resolved;
-                }
-
-                return FS::weakly_canonical(fsPath).string();
+                if(customWorkDir != nullptr)
+                    fsPath = ResolvePath(customWorkDir) / fsPath;
+                else
+                    fsPath = GetWorkDir() / fsPath;
             }
 
-            // expand virtual paths
-            FS::path resolved;
+            auto result = fsPath.string();
+            FilepathNormalize(result, false);
 
-            if (virtualPrefix == VPref::User) // user files location
+            // ModLoader support: keep game dir relative paths relative
+            if (StringStartsWith(GetWorkDir(), Filepath_Game, false))// curent work dir is game root
             {
-                resolved = GetUserDirectory();
+                FilepathRemoveParent(result, Filepath_Game); // remove game root path prefix

```

Comment:
remove this condition
`if (StringStartsWith(GetWorkDir(), Filepath_Game, false))`

we need to strip Filepath_Game from both WorkDir and CustomWorkDir (for mission loading)

Reply:
```
auto result = fsPath.string();
FilepathNormalize(result, false);
FilepathRemoveParent(result, Filepath_Game); // remove game root path prefix
```

Reply:
Hm maybe the GetWorkDir() should be replaced with cwd then. It for sure need be checked if stripped path in connection with work dir will still result in same location.

Reply:
let's not change this logic now, it's how it's done in .70 Just strip the path using new util

Reply:
Updated

Reply:
One last question. How do we process path to cleo_saves? From https://github.com/cleolibrary/CLEO5/issues/60#issuecomment-2354196794 it seems that changing cwd with game's internal function causes the path to be broken. Can we force it to be absolute with Cleo:\ prefix?

Reply:
This seems like problem within ML. RZL Trainer has similar problem. It checks for ini file, file not found so it performs whole initialization, creates the file in wrong location then proceeds to successfully read file from previous location that existed all the time. Like not all file APIs are hooked, or are not handled in same way.

That Skin Loader is changing CWD of the whole game so it also can be the cause as now reading relative path to cleo_saves (or any other path!) will fail.

Reply:
CLEO is using absolute paths when saving/loading state. I think that mod just tries to save it's ini file there.

Reply:
Ok let's investigate this separately